### PR TITLE
Use the Degreesign ° in gdalinfo reports of angles

### LINKED ISSUE #3763: Using the degree sign in gdalinfo
https://lists.osgeo.org/pipermail/gdal-dev/2021-April/054032.html
https://lists.osgeo.org/pipermail/gdal-dev/2021-April/054033.html

Now that PROJ data includes "AREA" descriptions with "°" the degree sign,
it would be nice if gdalinfo reported other angles with "°" too.

I am happy to do the work.

### PR DESCRIPTION
**Pull request by andrew-aitchison**
Now that PROJ data includes "AREA" descriptions with "°" the degree sign,
it would be nice if gdalinfo reported other angles with "°" too.
fixes #3763

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

Provide environment details, if relevant:

* OS:
* Compiler:

### CODE REVIEW

File: gdal/swig/perl/t/00.t [hunk 1]
```diff
@@ -62,7 +62,14 @@ for my $type (Geo::GDAL::DataTypes()) {
 # sub DecToPackedDMS ok
 
 my $dms = Geo::GDAL::DecToDMS(62, 'Long');
-ok($dms eq " 62d 0' 0.00\"E", "DecToDMS, got '$dms'"),
+ok($dms =~ " 62[°d] 0' 0.00\"E", "DecToDMS, got '$dms'"),
```
Comment (rouault) [inline gdal/swig/perl/t/00.t:65]: As Perl bindings will be removed in GDAL 3.5, hence those tests too, would you mind writing them in the Python autotest suite instead or in autotest/cpp ?

File: gdal/port/cpl_conv.cpp [hunk 2]
```diff
@@ -2225,6 +2228,11 @@ double CPLDMSToDec( const char *is )
             return tv;
         switch( *s )
         {
+          // UTF-8 "DEGREE SIGN" is 0xC2 0xB0 
+          case static_cast<char>(0xC2):
```
Comment (rouault) [inline gdal/port/cpl_conv.cpp:2232]: would be more elegant to do a switch ( static_cast<unsigned char>(*s) )  instead

File: gdal/port/cpl_conv.cpp [hunk 3]
```diff
@@ -2165,7 +2165,10 @@ static double proj_strtod(char *nptr, char **endptr)
     // Scan for characters which cause problems with VC++ strtod().
     while( (c = *cp) != '\0')
     {
-        if( c == 'd' || c == 'D' )
+        if( c == 'd' || c == 'D' ||
+            // UTF-8 "DEGREE SIGN" is 0xC2 0xB0 
+            c == static_cast<char>(0xC2) ||
```
Comment (rouault) [inline gdal/port/cpl_conv.cpp:2170]: More portable / less subjects to undefined/unspecified behaviour
```suggestion
            static_cast<unsigned char>(c) == 0xC2 ||
```

File: gdal/port/cpl_conv.cpp [hunk 4]
```diff
@@ -2165,7 +2165,10 @@ static double proj_strtod(char *nptr, char **endptr)
...
+            c == static_cast<char>(0xB0) )
```
Comment (rouault) [inline gdal/port/cpl_conv.cpp:2171]:
```suggestion
            static_cast<unsigned char>(c) == 0xB0 )
```

File: gdal/swig/perl/t/00.t (hunk 1 above)
Comment (andrew-aitchison) [reply gdal/swig/perl/t/00.t:65]: I'll have a go.
Python autotests are failing on my machine at
    from osgeo import gdal

Comment (rouault) [reply gdal/swig/perl/t/00.t:65]: maybe check that you haven't already "system" python bindings that interfer with. Try  "source scripts/setdevenv.sh" from gdal/

Comment (idanmiara) [reply gdal/swig/perl/t/00.t:65]: If you still have issues with Python then provide more details and I'll try to help.

Comment (andrew-aitchison) [reply gdal/swig/perl/t/00.t:65]: Thanks. I've made progress.
This is Ubuntu 21.04: the python 3 pytest is called pytest-3, so I was running a copy of the python2 version. Do we need to get configure to look for "pytest-3" ?

Currently pytest-3 is aborting with 
gcore/misc.py::test_misc_5 terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_M_create
Fatal Python error: Aborted

Full pytest-3 -vvs  output 
[pytest.errlog.20210503.txt](https://github.com/OSGeo/gdal/files/6415044/pytest.errlog.20210503.txt)

But I'm having a day off and not stuff yet.

Comment (rouault) [reply gdal/swig/perl/t/00.t:65]: not sure why you this error. you might try a "make clean" and rebuild : GDAL build scripts don't track header file dependencies

**Comment by rouault** [comment]
hum seeing https://github.com/OSGeo/PROJ/issues/2713 , I wonder if this will be a good idea (for our unlucky friends using Windows)

File: gdal/swig/perl/t/00.t (hunk 1 above)
Comment (andrew-aitchison) [reply gdal/swig/perl/t/00.t:65]: Yes, it was something like that.

**Comment by andrew-aitchison** [comment]
OK. If Windows doesn't display the degree symbol, I'll leave this for another year ...

What is the best way to leave this tidy in case we wish to come back to it some day ?
Do I close the pull request ?

**Comment by rouault** [comment]
> What is the best way to leave this tidy in case we wish to come back to it some day ?
> Do I close the pull request ?

yes, we can retrieve it by searching a bit

**Comment by jidanni** [comment]
Well you might as well close OSGeo/PROJ#2713 too, else there will be no way to distinguish it from those closed by the stalebot.

**Pull request closed without merging**